### PR TITLE
Add cpp linter

### PR DESCRIPTION
**Pull request by Geam**
It's mainly an copy/paste from the c one with few modification for cpp.

### CODE REVIEW

File: doc/ale.txt [hunk 1]
```diff
@@ -273,6 +275,16 @@ Default: `'-Wall'`
...
 
+-------------------------------------------------------------------------------
+4.4. cpp-gcc                                       *ale-linter-options-cpp-gcc*
+
+g:ale_cpp_gcc_options                                   *g:ale_cpp_gcc_options*
+
+Type: |String|
+Default: `'-Wall'`
+
+This variable can be change to modify flags given to gcc.
```
Comment (w0rp) [inline doc/ale.txt]: There's a typo here. They should be "changed" not "change."

**Comment by w0rp** [comment]
I guess those typos can be sorted out afterwards. :+1: to documenting the Fortran options. That's still undocumented. Both can be done in another PR.

**Comment by Geam** [comment]
Ok for C++ instead of Cpp in the doc and README. I've put cpp cause `set: filetype` give me cpp and I couldn't use ++ in the variable name (vim doesn't like it) but yes, it's wrong in the doc. 

For the parentheses, you mean in the README ? If yes, I think we should add a linter specific section under Options cause, if there is many linter for a language it can get messy. I will try something and if it's good enough, I will submit it.

**Comment by w0rp** [comment]
I was wondering if it could be written `C++ (filetype cpp)`, or maybe people will just know that anyway? Something like that.

**Comment by w0rp** [comment]
Yep, that'll do it. Cheers! :beers:

**Comment by Geam** [comment]
I've modify it to match the cython one

**Pull request merged**